tools/import_validation/import_validation.py

```diff
@@ -11,138 +11,161 @@
+            'MAX_DATE_LATEST':
+                (self.validator.validate_max_date_latest, 'stats'),
+            'MAX_DATE_CONSISTENT':
+                (self.validator.validate_max_date_consistent, 'stats'),
+            'DELETED_COUNT': (self.validator.validate_deleted_count, 'differ'),
+            'MODIFIED_COUNT':
+                (self.validator.validate_modified_count, 'differ'),
+            'ADDED_COUNT': (self.validator.validate_added_count, 'differ'),
+            'UNMODIFIED_COUNT':
+                (self.validator.validate_unmodified_count, 'differ'),
+            'NUM_PLACES_CONSISTENT':
+                (self.validator.validate_num_places_consistent, 'stats'),
+            'NUM_PLACES_COUNT':
+                (self.validator.validate_num_places_count, 'stats'),
+            'NUM_OBSERVATIONS_CHECK':
+                (self.validator.validate_num_observations_check, 'stats'),
+            'UNIT_CONSISTENCY_CHECK':
+                (self.validator.validate_unit_consistency, 'stats'),
+            'MIN_VALUE_CHECK':
+                (self.validator.validate_min_value_check, 'stats'),
+            'MAX_VALUE_CHECK':
+                (self.validator.validate_max_value_check, 'stats'),
+        }
 
     def run_validations(self) -> bool:
-        # Returns false if any validation fails.
-        status = True
-        with open(self.validation_output, mode='w',
-                  encoding='utf-8') as output_file:
-            output_file.write('test,status,message\n')
-            for config in self.validation_config:
-                result = self._run_validation(config)
-                # TODO: use CSV writer libs
-                output_file.write(
-                    f'{result.name},{result.status},{result.message}\n')
-                self.validation_result.append(result)
-                if result.status == 'FAILED':
-                    status = False
-        return status
+        """
+    Runs all validations specified in the config and returns the overall status.
+    """
+        overall_status = True
+        for config in self.validation_config:
+            validation_name = config['validation']
+            if validation_name not in self.validation_dispatch:
+                logging.warning('Unknown validation: %s', validation_name)
+                continue
+
+            validation_func, df_key = self.validation_dispatch[validation_name]
+            df = self.dataframes[df_key]
+
+            # Apply filters if they are defined in the config
+            if 'variableMeasured' in config:
+                df = filter_dataframe(df, config['variableMeasured'])
+
+            # Pass config to the validation function if it's needed
+            if validation_name in [
+                    'DELETED_COUNT', 'NUM_PLACES_COUNT',
+                    'NUM_OBSERVATIONS_CHECK', 'MIN_VALUE_CHECK',
+                    'MAX_VALUE_CHECK'
+            ]:
+                result = validation_func(df, config)
```

Comment:
would it be simpler to always pass in the config?
it may be useful to log the name, etc

Reply:
I changed it to pass the `config` for all invocations and eliminated this conditional block. As of now, I haven't included any logging statements in the `Validator` class. I deferred them to `ValidationRunner` instead. So some of them accept this param and do nothing, but it simplifies the code here.